pkg/kubefed2/federate/util.go

```diff
@@ -101,29 +103,37 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 			continue
 		}
 		for _, resource := range resourceList.APIResources {
+			var matchResource = func() {
+				if targetResource != nil {
+					multiMatched = true
+				} else {
+					targetResource = resource.DeepCopy()
+					targetResource.Group = gv.Group
+					targetResource.Version = gv.Version
+				}
+
+				matchedResources = append(matchedResources, groupQualifiedName(resource.Name, gv.Group))
+			}
+
 			if lowerKey == resource.Name ||
 				lowerKey == resource.SingularName ||
 				lowerKey == strings.ToLower(resource.Kind) ||
 				lowerKey == fmt.Sprintf("%s.%s", resource.Name, gv.Group) {
 
-				targetResource = &resource
-				break
+				matchResource()
+				continue
 			}
 			for _, shortName := range resource.ShortNames {
 				if lowerKey == strings.ToLower(shortName) {
-					targetResource = &resource
+					matchResource()
 					break
```

Comment:
This should be `continue` to ensure that duplicate short names are also detected (e.g. `deploy` amiguously refers to both `deployment.apps` and `deployment.extensions`).

Reply:
@marun this `break` just escapes the `resource.ShortNames` loop, doesn't it?

Reply:
For a resource, I guess it has at most one short name matched. 
deploy is a short name for both `deployment.apps` and `deployment.extensions`

Reply:
@yamt, yes, it breaks `shortNames` loop.